scripts/doindex/lib/task.mjs

```diff
@@ -0,0 +1,122 @@
+import { availableParallelism as cpuSize, cpus } from 'os';
+import { fork } from 'child_process';
+import { createResolve } from './index.mjs';
+
+/**
+ * @typedef {ChildProcess} Fork
+ * @property {number} [id]
+ * @property {boolean} [waits]
+ */
+
+export class FixedThreadPool {
```

Comment:
Anybody knows good library for thread pool management.

_kotlin coroutines should be nice_ 😁